packages/webapp/src/containers/Finances/index.jsx

```diff
@@ -121,293 +109,114 @@ class Finances extends Component {
-          moment(e.expense_date).isSameOrBefore(endDate, 'day')
-        ) {
-          total += Number(e.value);
-        }
-      }
-    }
-
-    return total;
-  };
-
-  getCropsByFieldID = (location_id) => {
-    const { managementPlans } = this.props;
-
-    let result = new Set();
-
-    for (let fc of managementPlans) {
-      if (fc.location_id === location_id) {
-        result.add(fc.crop_id);
-      }
-    }
-
-    return Array.from(result);
   };
 
-  convertToCropID = (final) => {
-    let fcIDs = Object.keys(final);
-    let result = {};
-
-    const { startDate, endDate } = this.state;
-    const { sales } = this.props;
-
-    for (let fkey of fcIDs) {
-      let value = final[fkey];
-      if (result.hasOwnProperty(value.crop_id)) {
-        result[value.crop_id].profit += value.profit;
-      } else {
-        result[value.crop_id] = {
-          crop: this.props.t(`crop:${value.crop_translation_key}`),
-          location_id: value.location_id,
-          crop_id: value.crop_id,
-          profit: value.profit,
-        };
-      }
-    }
-
-    //apply sales
-    for (let sale of sales || []) {
-      const saleDate = moment(sale.sale_date);
-
-      if (saleDate.isSameOrAfter(startDate, 'day') && saleDate.isSameOrBefore(endDate, 'day')) {
-        for (let cp of sale.cropSale) {
-          if (cp.crop && result.hasOwnProperty(cp.crop.crop_id)) {
-            result[cp.crop.crop_id].profit += Number(cp.sale_value);
-          } else {
-            result[cp.crop.crop_id] = {
-              crop: this.props.t(`crop:${cp.crop.crop_translation_key}`),
-              location_id: 'not available',
-              crop_id: cp.crop.crop_id,
-              profit: Number(cp.sale_value),
-            };
-          }
-        }
-      }
-    }
-
-    return result;
-  };
+  const totalRevenue = calcActualRevenue(sales, startDate, endDate).toFixed(2);
+  const estimatedRevenue = getEstimatedRevenue(managementPlans);
+  const labourExpense = roundToTwoDecimal(calcTotalLabour(tasks, startDate, endDate));
+  const otherExpense = calcOtherExpense(expenses, startDate, endDate);
+  const totalExpense = (parseFloat(otherExpense) + parseFloat(labourExpense)).toFixed(2);
+
+  return (
+    <div className={styles.financesContainer}>
+      <Title style={{ marginBottom: '8px' }}>{t('SALE.FINANCES.TITLE')}</Title>
+      <hr />
+      <Semibold style={{ marginBottom: '8px' }}>{t('SALE.FINANCES.ACTION')}</Semibold>
+
+      <div className={styles.buttonContainer}>
+        <Button
+          style={{ height: '48px' }}
+          onClick={() => {
+            dispatch(setPersistedPaths(['/expense_categories', '/add_expense']));
+            history.push('/expense_categories');
+          }}
+          color="success"
+        >
+          {t('SALE.FINANCES.ADD_NEW_EXPENSE')}
+        </Button>
+        <Button
+          style={{ height: '48px' }}
+          onClick={() => {
+            history.push('add_sale');
```

Comment:
It looks like the newest changes are lost here?
```
this.props.dispatch(
  setPersistedPaths(['/revenue_types', '/add_sale', '/manage_custom_revenues']),
);
history.push('/revenue_types');
```

Reply:
Thank you! 🙏 I should I have been more careful with that merge conflict.